docs/api/types.rst

```diff
@@ -0,0 +1,423 @@
+| ``float32``,         | ``number``                                          |
+| ``float64``          |                                                     |
+| ``int16``,           |                                                     |
+| ``int32``,           |                                                     |
+| ``int64``            |                                                     |
++----------------------+-----------------------------------------------------+
+| ``decimal``          | n/a                                                 |
++----------------------+-----------------------------------------------------+
+| ``json``             | ``string``                                          |
++----------------------+-----------------------------------------------------+
+| ``uuid``             | ``edgedb.UUID``                                     |
++----------------------+-----------------------------------------------------+
+
+.. note::
+
+    Inexact single-precision ``float`` values may have a different
+    representation when decoded into a JavaScript number.  This is inherent
+    to the implementation of limited-precision floating point types.
+    If you need the decimal representation to match, cast the expression
+    to ``float64`` in your query.
+
+.. note::
+
+    Due to precision limitations the ``decimal`` type cannot be decoded to a
+    JavaScript number. Use an explicit cast to ``float64`` if the precision
+    degradation is acceptable or a cast to ``str`` for an exact decimal
+    representation.
+
+
+.. _edgedb-js-types-set:
+
+Sets
+====
+
+.. js:class:: Set() extends Array
+
+    ``Set`` represents the set of values returned by a query. If a query
+    contained an explicit ``ORDER BY`` clause, the values will be ordered,
+    otherwise no specific ordering is guaranteed.
+
+    This type also allows to differentiate between a set of values and an
+    explicit array.
+
+
+Arrays
+======
+
+EdgeDB ``array``  maps onto the JavaScript ``Array``.
+
+.. code-block:: js
+
+    // Use the Node.js assert library to test results.
+    const assert = require("assert");
+    const edgedb = require("edgedb");
+    const connect = edgedb.default;
+    var conn;
+
+    async function main() {
+      conn = await connect({
+        dsn: "edgedb://edgedb@localhost/"
+      });
+
+      let data = await conn.fetchOne("SELECT [1, 2, 3]");
+
+      // The result is an Array.
+      assert(data instanceof Array);
+      assert(typeof data[0] === "number");
+      assert(data.length === 3);
+      assert(data[2] === 3);
+    }
+
+    main().finally(() => {
+      conn.close();
+    });
+
+
+.. _edgedb-js-types-object:
+
+Objects
+=======
+
+``Object`` represents an object instance returned from a query. The value of an
+object property or a link can be accessed through a corresponding object key:
+
+.. code-block:: js
+
+    // Use the Node.js assert library to test results.
+    const assert = require("assert");
```

Comment:
Can we use `import` statement?

Reply:
`import` is not supported in plain nodejs without `--experimental-modules` and using `*.mjs` files. The bottom line is that the import syntax requires the use of some extra things like babel or special node mode.

I have, however, changes the package to expose `connect` as both a default and a named export so that `const {connect} = require("edgedb");` should work (currently it doesn't).